main.go

```diff
@@ -98,6 +98,12 @@ func main() {
 
 	config.Fprint(verbose)
 
+	// use kubeclient to check the current namespace
+	namespace, _ := k8s.CurrentNamespace()
+	if namespace == "kube-system" {
```

Comment:
This isn't quite the error we were running into.

You probably can run the openfaas core components in the kube-system namespace.

It was that you can't deploy functions there.

So we already have validation in the deployment handlers, but we could do better by checking the functionNamespace variable (if set), and exiting upon startup

Reply:
i thought this was part of the suggestion https://github.com/openfaas/faas-netes/issues/1022#issuecomment-1253906489

but i see that what we also need to do is provide a meaningful api error as well